src/ui/textfield.rs

```diff
@@ -1,15 +1,63 @@
 //! A fairly simplistic text field implementation.
 
+use std::{borrow::BorrowMut, ops::Range};
+
+use copypasta::{ClipboardContext, ClipboardProvider};
 use skulpin::skia_safe::*;
 
 use crate::ui::*;
 
+/// Text field selection struct
```

Comment:
I think this comment could be a bit more descriptive. Maybe try

> Text field selection.
> 
> This stores two cursors: the selection anchor, and the actual text cursor.
These cursors are modified appropriately as the user edits text.